src/FSharp.Core/array.fs

```diff
@@ -1588,6 +1588,22 @@ module Array =
 
         acc
 
+    [<CompiledName("Sum")>]
+    let inline sumFloat (array: float array) : float =
+        System.Linq.Enumerable.Sum array
+
+    [<CompiledName("Sum")>]
+    let inline sumFloat32 (array: float32 array) : float32 =
+        System.Linq.Enumerable.Sum array
+
+    [<CompiledName("Sum")>]
+    let inline sumInt (array: int array) : int =
+        System.Linq.Enumerable.Sum array
+
+    [<CompiledName("Sum")>]
+    let inline sumInt64 (array: int64 array) : int64 =
+        System.Linq.Enumerable.Sum array
```

Comment:
I would think that this would be a reasonable place to use [static optimization syntax](https://github.com/dotnet/fsharp/blob/9b4a7ce7bfdb7b5c9e50bee9b9c084111e5078a0/src/FSharp.Core/prim-types.fs#L5358-L5405) to specify which types should delegate to the LINQ method and which to the existing code, e.g.,

```fsharp
let inline sum (array: ^T array) : ^T =
    existingSumCode array
    when ^T : int   = System.Linq.Enumerable.Sum array
    when ^T : int64 = System.Linq.Enumerable.Sum array
    …
```

Reply:
Sure. I expect "static optimization conditionals" are a compile-time thing and not runtime? Because I can't check easily with sharplab.io, it says "error FS0817: Static optimization conditionals are only for use within the F# library"

Reply:
Yes, the appropriate implementation would be chosen at compile-time once the type parameter was resolved.

You could add some IL tests under https://github.com/dotnet/fsharp/tree/main/tests/FSharp.Compiler.ComponentTests/EmittedIL if you wanted.